src/rgw/rgw_crypt.cc

```diff
@@ -359,25 +359,26 @@ class AES_256_CBC : public BlockCrypt {
                      const unsigned char (&key)[AES_256_KEYSIZE],
                      bool encrypt)
   {
-    static std::atomic<bool> failed_to_get_crypto(false);
-    CryptoAccelRef crypto_accel;
-    if (! failed_to_get_crypto.load())
+    struct CryptoAccelRefWrapper
     {
-      crypto_accel = get_crypto_accel(cct);
-      if (!crypto_accel)
-        failed_to_get_crypto = true;
-    }
+      CryptoAccelRef crypto_accel;
+      CryptoAccelRefWrapper(CephContext* cct)
+      : crypto_accel(get_crypto_accel(cct)) {}
+      ~CryptoAccelRefWrapper() { crypto_accel = nullptr;}
+    };
+    CryptoAccelRefWrapper* ref;
+    cct->lookup_or_create_singleton_object(ref, "CBC_transform_accel");
```

Comment:
i thought this use of `lookup_or_create_singleton_object()` was to replace something that was previously declared `static`. is this just preparation for something in https://github.com/ceph/ceph/pull/14498?